template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/CmTemplateProcessor.java

```diff
@@ -76,9 +79,13 @@
 import com.sequenceiq.cloudbreak.template.processor.configuration.HostgroupConfigurations;
 import com.sequenceiq.cloudbreak.template.processor.configuration.SiteConfigurations;
 import com.sequenceiq.cloudbreak.template.views.CustomConfigurationPropertyView;
+import com.sequenceiq.cloudbreak.view.InstanceMetadataView;
 
 public class CmTemplateProcessor implements BlueprintTextProcessor {
 
+    private static final Map<String, String> ROLE_DEPENDENCY = Map.ofEntries(
```

Comment:
Move this, and methods related to determining the dependency into a separate ComponentDependenyService. This Class should have JavaDocs explaining the intent, and very specifically - "ROLE_DEPENDENCIES is not an exhaustive list of dependencies. The absence of a dependency means that the dependency is not yet defined, and should be handled accordingly."